docker/Dockerfile-prod

```diff
@@ -15,13 +37,32 @@ RUN chown app:app /db
 COPY migrations /migrations
 
 COPY requirements.txt /requirements.txt
-RUN pip install --no-cache -r /requirements.txt
+
+# Install pure-Python, compiled, and OS package dependencies.  Use scanelf to
+# uninstall any compile-time OS package dependencies and keep only the run-time
+# OS package dependencies.
+RUN set -ex \
+    && apk add --no-cache --virtual .build-deps \
+           gcc \
+           libc-dev \
+           musl-dev \
+           linux-headers \
+           pcre-dev \
+    && pip install --no-cache -r /requirements.txt \
+	&& runDeps="$( \
+		scanelf --needed --nobanner --recursive /usr/local \
+			| awk '{ gsub(/,/, "\nso:", $2); print "so:" $2 }' \
+			| sort -u \
+			| xargs -r apk info --installed \
+			| sort -u \
+	)" \
+	&& apk add --virtual .python-rundeps $runDeps \
```

Comment:
The list of packages changes every time we add a new compiled dependency.  So far that's been for database adapters, uWSGI, and crypto/SSL handling.

This is the second time a reviewer has said they would be less surprised by an explicit list of runtime packages, and I'm sensing a trend, so I'll look into changing it.  :)  But we might not want to block on the rewrite and make that a follow-up commit.